aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1908,7 +1911,84 @@ Tensor& vdot_out(const Tensor& self, const Tensor& other, Tensor& result) {
   return result.fill_(self.vdot(other));
 }
 
-static bool should_fold(const Tensor& tensor1, const Tensor& tensor2) {
+#define AUTOTUNE_MAX_DIM 4
+
+struct MatmulAutotuneCacheKey {
+  int tensor1_dim[AUTOTUNE_MAX_DIM];
```

Comment:
The type of elements of Tensor.sizes() is `int64_t`. You could also use a `DimVector` from `c10/util/DimVector.h`.

Reply:
Will go with `int64_t` as the params hashing complains that `DimVector` isn't POD